thonny/code.py

```diff
@@ -157,6 +158,9 @@ def _load_file(self, filename, keep_undo=False):
         with tokenize.open(filename) as fp:  # TODO: support also text files
             source = fp.read()
 
+        # get the file file format (DOS/UNIX)
+        self._newlines = fp.newlines
```

Comment:
Good find! I did not know about this attribute. 

I would expect this to be under `with`, but apparently closing the file pointer (which happens after exiting `with` block), leaves `newlines` property intact.